tools/twitter-api/scrolling.py

```diff
@@ -48,44 +44,41 @@ def validateTweets(tweets):
   return True
  
 # method for continuously searching for all tweets
-def search(max_hits=999999999):
+def search(es_index, max_hits=None):
   print('###########################')
   print('        Begin Search       ')
   print('###########################')
   response = []
- 
+
   # perform a search for 2ms and get initial index
   page = es.search(index=es_index,
                        scroll='2m',
                        size=1000,
                        body=search_param)
   scroll_id = page['_scroll_id']
+  ids = [scroll_id]
   scroll_size = page['hits']['total']['value']
   response.extend(page['hits']['hits'])
   print('%d Tweets Gathered' % (len(response)))
  
   # Start scrolling until we have all documents that match our query
   while (scroll_size > 0):
-    if (len(response) >= max_hits):
+    if (max_hits != None and len(response) >= max_hits):
       break
     page = es.scroll(scroll_id=scroll_id, scroll='2m')
     # Update the scroll ID
     scroll_id = page['_scroll_id']
+    ids.append(scroll_id)
     # Get the number of results that we returned in the last scroll
     validateTweets(page['hits']['hits'])
     scroll_size = len(page['hits']['hits'])
     response.extend(page['hits']['hits'])
     print('%d Tweets Gathered' % (len(response)))
+  
+  es.clear_scroll(body=None, scroll_id=",".join(ids))
   return response
  
 if __name__ == '__main__':
-  response = search(max_hits=1000)
+  response = search("coronavirus-data-all", max_hits=1000)
```

Comment:
Shouldn't this pass client_config["es_index"]?

Reply:
In this particular case yes, it was used for debugging purposes. But you do bring up a valid point. The function `search` now takes in an index to scroll over. This should be specified / configured in `twitter_api_handler.py`